astroid/rebuilder.py

```diff
@@ -88,9 +91,13 @@ class TreeRebuilder:
     """Rebuilds the _ast tree to become an Astroid tree"""
 
     def __init__(
-        self, manager: AstroidManager, parser_module: Optional[ParserModule] = None
+        self,
+        manager: AstroidManager,
+        parser_module: Optional[ParserModule] = None,
+        data: Optional[str] = None,
```

Comment:
True, inside `astroid` it will always be `str`. Technically, you could consider `TreeRebuilder` part of the public interface however. Adding a new required argument would then be a breaking change.

Defining it as `Optional` isn't too bad considering everything. The `name_node` / `position` needs to be `Optional`, so a small check `if not self._data` doesn't really hurt.